presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -284,36 +283,88 @@ private void checkEncodeInputValidity(List<Integer> input, int maximumBits)
         checkArgument((totalBitLength <= maximumBits), format("The z-address type specified is not large enough to hold %d bits.", totalBitLength));
     }
 
+    /**
+     * The BoundedZValueRange class contains non-null Integer minimum and maximum value lists. The bounds are all inclusive.
+     */
+    private static class BoundedZValueRange
+    {
+        private final List<Integer> minimumValues;
+        private final List<Integer> maximumValues;
+
+        public BoundedZValueRange(List<Integer> minimumValues, List<Integer> maximumValues)
+        {
+            this.minimumValues = minimumValues;
+            this.maximumValues = maximumValues;
+        }
+
+        public Integer getMinimumValueAt(int index)
+        {
+            return minimumValues.get(index);
+        }
+
+        public Integer getMaximumValueAt(int index)
```

Comment:
do we expect to return nulls? If not, let's use `int`. The rule of thumb is to use primitive types whenever possible. If nulls are needed, use OptionalXXX. Boxed types are relatively expensive for low-level libraries.